CookingBook/src/networking/Client/Client.cpp

```diff
@@ -0,0 +1,218 @@
+#include "Client.h"
+
+Client::Client() :
+	sIpAddress(""),
+	iPort(0),
+	ClientSocket(INVALID_SOCKET),
+	Time(std::time(0)),
+	CurrentTime(std::localtime(&Time)){
+	ZeroMemory(&ClientInfo, sizeof(ClientInfo));
+	iClientInfoSize = sizeof(ClientInfo);
+	ClientLogFile.open(sClientLogFileName,std::ios::binary|std::ios::app);
+}
+
+Client::~Client() {
+	WSACleanup();
+	closesocket(ClientSocket);
+	if(ClientLogFile.is_open()){
+		ClientLogFile.close();
+	}
+}
+
+void Client::LogInfo(const std::string& sInfo){
+	ClientLogFile<<"["<<CurrentTime->tm_year+1900<<":"<<
+	CurrentTime->tm_mon+1<<":"<<
+	CurrentTime->tm_mday<<":"<<
+	CurrentTime->tm_hour<<":"<<
+    CurrentTime->tm_min<<":"<<
+	CurrentTime->tm_sec<<"]"<<" "+sInfo<<std::endl;
+}
+
+void Client::SetClient(const std::string& sIpAddress, const int iPort){
+	this->sIpAddress = sIpAddress;
+	this->iPort = iPort;
+	IPAddressFile.open(sIPAddressFileName,std::ios::binary|std::ios::out|std::ios::trunc);
+	IPAddressFile<<sIpAddress<<std::endl;
+	IPAddressFile<<iPort;
+	IPAddressFile.close();
+}
+
+bool Client::CheckIP(){
+	IPAddressFile.open(sIPAddressFileName,std::ios::ate|std::ios::binary);
+	bool bResult;
+	if(IPAddressFile.is_open()){
+		if(IPAddressFile.tellg()==0){
```

Comment:
Return Value
The current position in the stream.
If either the stream buffer associated to the stream does not support the operation, or if it fails, the function returns -1.
Why are you comparing with zero?